### ISSUE TITLE
Parallel option not working

### ISSUE THREAD

**Issue by halhenke** [lukeed/taskr#73]
Running a Flyfile with

``` javascript
export default function* () {
  yield this.start(["babel", "jade", "sass"], { parallel: true })
}
```

gives the following errors in the console

```
06:50:28] Starting "default"
[06:50:28] Starting "babel"
[06:50:28] Starting "jade"
[06:50:28] Starting "sass"
[06:50:28] babel failed due to TypeError: undefined is not iterable!
[06:50:28] jade failed due to TypeError: undefined is not iterable!
[06:50:28] sass failed due to TypeError: undefined is not iterable!
[06:50:28] Finished "default" in 48 ms
```

So i imagine this stems from here:

``` javascript
      if (parallel) {
        yield tasks.map((task) =>
          this.exec(task, value, Object.create(this)))
      }
```

as opposed to the execution in series:

``` javascript
      else
        for (let task of tasks)
          value = yield this.exec(task, value)
      return value
```

Have fiddled a bit (thought moving yield inside the map function would work) but havent resolved the issue - assuming this isnt, of course, just something on my end.

**Comment by ghost** [comment]
@halhenke I was able to reproduce the bug easily. Nice catch and thanks for reporting this. 

I was able to find the culprit by running `fly` with `DEBUG=fly:*` and printing the value of `this` inside the `babel task` in the flyfile. The problem is that plugins are not rebound the new copy `Object.create(this)`. So, when `sass`, `jade` or `babel` runs, each call `Fly.proto.filter` which runs a closure **bound** to the main `fly` instance, not to the concurrent copy!

This is a silly bug and there are few approaches to solve this. For example delaying plugin load until we have the fly instance we want.

**Comment by ghost** [comment]
To make clear, your code is fine, this is a bug in fly. The following is the example `flyfile` I am using.

``` js
export default function* () {
  this.clear("dist")
  yield this.start(["babel", "sass", "jade"], { parallel: true })
}

export function* babel () {
  yield this
    .source("src/*.js")
    .babel({ stage: 0 })
    .target("dist")
}

export function* sass () {
  yield this
    .source("src/*.scss")
    .sass({ outputStyle: "compressed" })
    .target("dist")
}

export function* jade () {
  yield this
    .source("src/*.jade")
    .jade()
    .target("dist")
}
```

I will be patching a fix for this in the next release :facepunch:

**Comment by halhenke** [comment]
Cool, glad i could provide some help.

Wish i could help patch this right now but I still dont grok the `co` library and how everything fits together enough :-(

**Comment by ghost** [comment]
You can patch `Fly.proto.start` temporarily like this:

The real patch will be something else. 

``` js
//...
          const fly = Object.create(this)
          this.plugins.forEach(({ name, plugin }) => {
            _("(load) %o", name)
            plugin.call(fly, debug(name.replace("-", ":")))
          })
          return this.exec(task, value, fly)
        })
//...

```

**Comment by ghost** [comment]
Fixed in [@v0.5.0](https://github.com/flyjs/fly/releases/tag/v0.5.0)
